src/lib/notion/announcements.ts

```diff
@@ -15,27 +18,73 @@ export async function getAnnouncementsPageIds() {
   return await getPageIds(ANNOUNCEMENTS_DATABASE_ID);
 }
 
-export async function getAnnouncements() {
+export async function getFeaturedAnnouncements() {
+  return await getAnnouncements({ featuredOnly: true });
+}
+
+export async function getAnnouncements(options?: { featuredOnly: boolean }) {
+  const featuredOnly = options?.featuredOnly ?? false;
+
   const announcementsPageIds = await getAnnouncementsPageIds();
   const announcements: AnnouncementDataType[] = [];
   for (const announcementPageId of announcementsPageIds) {
     const page = (await notion.pages.retrieve({ page_id: announcementPageId })) as any;
-    const title = page.properties.Name.title[0].plain_text;
-    const callout = page.properties.Callout.rich_text;
-    const latest = page.properties['Latest Callout'].checkbox;
+
+    if (featuredOnly && !page.properties.Featured.checkbox) {
+      continue;
+    }
 
     announcements.push({
-      announcementPageId: announcementPageId,
-      title,
-      callout,
-      latest,
+      ...getAnnouncementPageProperties(page, announcementPageId),
     });
   }
   return announcements;
 }
 
+export async function getNewsPageBySlug(slug: string) {
+  const newsPage = await getPageBySlug(ANNOUNCEMENTS_DATABASE_ID, slug);
+
+  if (!newsPage) return undefined;
+
+  const newsPageId = newsPage.id;
+
+  return { newsPageId, ...getAnnouncementPageProperties(newsPage, newsPageId) };
+}
+
+function getAnnouncementPageProperties(page: any, pageId: string) {
+  return {
+    announcementPageId: pageId,
+    slug: page.properties.Slug.rich_text[0]?.plain_text || pageId,
+    title: page.properties.Name.title[0].text.content,
+    description: page.properties.Description.rich_text[0]?.text.content,
+    homePageURL: page.properties['Cover URL'].rich_text[0]?.plain_text || '/default',
+    callout: page.properties.Callout.rich_text,
+    latest: page.properties['Latest Callout'].checkbox,
+  };
+}
+
 export async function getLatestCallout() {
-  const announcements = await getAnnouncements();
-  // finds first announcement in array with Latest Callout property checked and non-empty Callout property
-  return announcements.find(a => a.latest && a.callout.length);
```

Comment:
Good that we are avoiding O(n) linear search and using @hellomusa 's DB query approach